src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -33,7 +33,7 @@ internal abstract class ODataWriterCore : ODataWriter, IODataOutputInStreamError
         private readonly bool writingResourceSet;
 
         /// <summary>True if the writer was created for writing a delta response; false otherwise.</summary>
-        private readonly bool writingDelta;
+        private readonly Stack<bool> writingDelta = new Stack<bool>();

```

Comment:
Why cannot we put the status into the Scope?

Reply:
We probably could, I was just minimizing the change for this PR, with the knowledge that this is a good candidate for clean-up in 8.0.